lib/frontdesk.rb

```diff
@@ -0,0 +1,85 @@
+require 'date'
+
+require_relative 'date.rb'
+require_relative 'reservation.rb'
+
+module Hotel
+  class FrontDesk
+    
+    attr_reader :all_reservations, :rooms
+    
+    def initialize
+      @all_reservations = all_reservations || []
+      @rooms = [*1..20]
+    end
+
+    def assign_room(new_reservation)
+      taken_rooms = []
+      all_rooms = @rooms.dup
+      @all_reservations.each do |reservation|
+        reservation.no_conflict?(new_reservation.start_date, new_reservation.end_date) ? taken_rooms << reservation.assigned_room : next
+      end
+      room_to_assign = (all_rooms - taken_rooms.flatten)
+      assigning = room_to_assign.sample(new_reservation.num_rooms)
+      taken_rooms.include?(assigning)
+        assigning = room_to_assign.sample(new_reservation.num_rooms)
+    end
```

Comment:
You can simplify this using `reject`:

```suggestion
    def assign_room(new_reservation)
      return @all_reservations.reject do |reservation|
        reservation.no_conflict?(new_reservation.start_date, new_reservation.end_date)
      end.sample
    end
```

Reply:
Also, at least reading this it looks like you're rejecting all of the rooms that _don't_ conflict which confuses me.